### PR TITLE
Tailor's kit update + balancing

### PR DESCRIPTION
**Pull request by Coolthulhu**
Tailor's kit had two big problems: balance and UI.

First, it was simply way too strong. A lucky drop of tailor's kit could allow a day 0 survivor to shrug off brute smashes with no damage, turning a jumpsuit into something closer to a survivor suit than clothing. And survivor suit is already overpowered enough at 6 tailoring.
Nerfed it by ensuring that armor modified with leather or Kevlar always has some encumbrance. Doesn't prevent making awesome armor, but makes it moderately bulky.

Problems with UI were:
- No way to preview changes
- No way to tell if you have enough materials until "you don't have enough x"
- Having 2 mods would display as "(W) (F)", which takes a lot of space
  Changed the UI to:
- Preview changes (for example, "Add kevlar. Bash/Cut: 5/5->10/12")
- Gray out options that you don't have enough materials for
- Make the item name wrap all mods together like "(WFLK)"

Other changes:
- Allow removal of mods. No refunds, query to make sure player didn't misclick and knows the no refunds part
- Greatly buffed thread costs. Still not big, just not free
- Removed the 2 mods cap. It added nothing to the game, as items are generally either made for warmth or defense, not both. And now display handles multiple mods more nicely, so long name is not a problem. To compensate, increased difficulty of installing second (and third and fourth) mod.
- Increased the difficulty, so that a total newbie tailor can't just make armor out of his t-shirt
- Increased skill practice, made it scale with modified item's volume (and thus material cost). May need tweaking as I didn't add any cap and leather is cheap, so you could just repeatedly armor hazmat with leather
- Improved code quality of the function by a factor of 300. It still could use an improvement, though.

### CODE REVIEW

**Comment by Zireael07** [comment]
Great!!!!

File: src/iuse.cpp [hunk 1]
```diff
@@ -2661,39 +2656,28 @@ int iuse::sew_advanced(player *p, item *it, bool, const tripoint& )
...
+    const auto add_material = [&]( const itype_id &material,
+                                   const itype_id &mat_item,
+                                   const std::string &plural ) {
+        if( mod->made_of( material ) ) {
+            repair_items.push_back( mat_item );
+            plurals.push_back( rm_prefix( _( plural.c_str() ) ) );
+        }
+    };
+
+    add_material( "cotton", "rag", "<plural>rags" );
```
Comment (BevapDin) [inline src/iuse.cpp]: The `_()` needs to be around the literal string, otherwise it will not be extracted for translation.

File: src/iuse.cpp [hunk 2]
```diff
@@ -2704,322 +2688,157 @@ int iuse::sew_advanced(player *p, item *it, bool, const tripoint& )
...
+    temp_item = modded_copy( mod, "kevlar_padded" );
+    enab = can_add_mod( "kevlar_padded", "kevlar_plate" );
+    tmenu.addentry( 3, enab, MENU_AUTOASSIGN, _("%s (Bash/Cut: %d/%d->%d/%d, Encumbrance: %d->%d)"),
+        mod->item_tags.count("kevlar_padded") == 0 ? _("Pad with Kevlar") :
+                                                     _("Destroy Kevlar padding"),
+        mod->bash_resist(), mod->cut_resist(), temp_item.bash_resist(), temp_item.cut_resist(),
+        mod->get_encumber(), temp_item.get_encumber() );
+
+    tmenu.addentry( 4, true, 'q', _("Repair/fit in") );
+    tmenu.addentry( 5, true, 'q', _("Cancel") );
```
Comment (BevapDin) [inline src/iuse.cpp]: Both entries have the same shortcut.

**Comment by PropaneSoup** [comment]
These all look like very nice changes. I'm not totally sold on the "WFLK" attached to the item name, simply because it looks a little strange with them all strung together. Though to keep the names short I suppose it is for the best.
You may want to take a look at increasing the time to work on the clothing as well, scaling it with the clothing volume like materials needed and exp gain does, as 5 turns is quick to potential get a lot of exp.

**Comment by DavidKeaton** [comment]
why not display M for modified (or something), and have the details in the item info?

File: src/item.cpp [hunk 3]
```diff
@@ -1897,18 +1897,30 @@ std::string item::tname( unsigned int quantity, bool with_prefix ) const
...
         }
     }
+    // Wrap all armor mods together
+    const bool has_armor_mod = item_tags.count("wooled") + item_tags.count("furred") +
+        item_tags.count("leather_padded") + item_tags.count("kevlar_padded") > 0;
+    if( has_armor_mod ) {
+        ret << " (";
+    }
+    // It's probably hard to translate a single letter
+    // Any good way of doing _("W")[0] that aren't as hacky as _("W")[0]?
```
Comment (kevingranade) [inline src/item.cpp]: How about just item_name + like with weapon mods, and only display the details when examining the item?

Comment (Zireael07) [inline src/item.cpp]: Good point.

**Comment by DavidKeaton** [comment]
@kevingranade great minds think alike! :-)

**Comment by Coolthulhu** [comment]
Changed the mods to add a single `+` to name, like with guns except always just 1.
Only 1, because I recall people not liking the multiple pluses on guns, as they can make the name grow huge. We currently have only 4 mods and `++++` isn't that long, but if we ever wanted more, we'd have the same problems as with guns.

**Comment by DavidKeaton** [comment]
one thing is making the plus signs different colors and showing only one of each.

**Pull request merged**